### PR TITLE
Add models for the `lastaflute` framework

### PR DESCRIPTION
**Pull request by egregius313**
Adds some models for the `org.lastaflute.web` framework.

Methods annotated with the `org.lastaflute.web.Execute` annotation are handlers for URLs, so the parameters of the methods annotated with the `org.lastaflute.web.Execute` annotation are likely either URL parameters or forms.

### CODE REVIEW

**Comment by egregius313** [comment]
Test faillures appear to be provenance numbering 🙄

**Comment by owen-mc** [comment]
Is there any good documentation of LastaFlute? Also, do you think it would be possible/beneficial to restrict the types of parameters to methods with an `@Execute` annotation which we consider as sources? Or do you think any such parameter should be a source?

**Comment by egregius313** [comment]
> Is there any good documentation on LastaFlute?

There's https://dbflute.seasar.org/lastaflute/. A few pages are in English, but many are in Japanese. The project has JavaDocs, but I can't find a copy online (had to build one locally).

> Also, do you think it would be possible/beneficial to restrict the types of parameters to methods with an @Execute annotation which we consider as sources? Or do you think any such parameter should be a source?

We could easily enough filter out things whose type would make it a `SimpleTypeSanitizer`. But I think it should be fine to treat all of the parameters as sources and have the individual queries decide on their filtration.

File: java/ql/lib/semmle/code/java/frameworks/Lastaflute.qll [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+/**
+ * The parameter of a method defining a URL handler using the Lastaflute framework.
+ */
+class LastafluteHandlerParameterSource extends RemoteFlowSource {
+  LastafluteHandlerParameterSource() {
+    exists(Parameter p | p.getCallable().getAnAnnotation() instanceof LastafluteExecuteAnnotation |
+      p = this.asParameter()
+    )
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/frameworks/Lastaflute.qll:22]: I suggest writing this using MaD:
```
- ["org.lastaflute.web", "Execute", False, "", "", "Annotated", "Parameter", "remote", "manual"]
```

Comment (aschackmull) [reply java/ql/lib/semmle/code/java/frameworks/Lastaflute.qll:22]: See e.g. the qltests in `java/ql/test/library-tests/dataflow/external-models/` to see what's possible using the `"Annotated"` functionality of MaD.

**Review by owen-mc — APPROVED** [review]

**Pull request merged**